pkg/frontend/query_result.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package frontend
+
+import (
+	"github.com/google/uuid"
+	"github.com/matrixorigin/matrixone/pkg/catalog"
+	"github.com/matrixorigin/matrixone/pkg/common/mpool"
+	"github.com/matrixorigin/matrixone/pkg/container/batch"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/objectio"
+	"strings"
+)
+
+func openSaveQueryResult(ses *Session) bool {
+	// TODO: Graceful judgment
+	sql := strings.ToLower(ses.sql)
+	if strings.Contains(sql, "meta_scan") || strings.Contains(sql, "result_scan") || ses.tStmt == nil {
```

Comment:
this test is strange.   there maybe a user table/column called result scan.    And, actually it is perfectly reasonable to cache a query result that say, select/join/ using a previous result.   You may want to cache this too.   A better way is to check ses.tStmt on its tree structure.   We just dont want to cache a simple select */proj list from result_scan.

Reply:
done